sql/table.cc

```diff
@@ -5227,9 +5227,15 @@ int TABLE::verify_constraints(bool ignore_failure)
       /* yes! NULL is ok, see 4.23.3.4 Table check constraints, part 2, SQL:2016 */
       if ((*chk)->expr->val_int() == 0 && !(*chk)->expr->null_value)
       {
-        my_error(ER_CONSTRAINT_FAILED,
-                 MYF(ignore_failure ? ME_JUST_WARNING : 0), (*chk)->name.str,
-                 s->db.str, s->table_name.str);
+        if ((*(*chk)).get_vcol_type()==VCOL_CHECK_TABLE)
+          my_error(ER_CONSTRAINT_FAILED,
+                   MYF(ignore_failure ? ME_JUST_WARNING : 0), (*chk)->name.str,
+                   s->db.str, s->table_name.str);
+        else // vcol_type=vcol_check_field
```

Comment:
One should make sure this always hold by setting a DEBUG assert, not in a comment.

Also, I don't like the code duplication. Let's do this:

```
        enum_vcol_info_type vcol_type= (*chk)->get_vcol_type();
        DBUG_ASSERT(vcol_type == VCOL_CHECK_TABLE ||
                    vcol_type == VCOL_CHECK_FIELD);
        my_error(vcol_type == VCOL_CHECK_TABLE ? ER_CONSTRAINT_FAILED :
                                                 ER_FIELD_CONSTRAINT_FAILED,
                 MYF(ignore_failure ? ME_JUST_WARNING : 0), (*chk)->name.str,
                 s->db.str, s->table_name.str);
```